File: qqrobot-spring-boot-starter/src/main/java/de/honoka/qqrobot/starter/config/property/LoggerProperties.java
package de.honoka.qqrobot.starter.config.property;

import de.honoka.sdk.util.file.FileUtils;
import lombok.Data;
import org.springframework.boot.context.properties.ConfigurationProperties;

import java.nio.file.Paths;

@ConfigurationProperties("honoka.qqrobot.logger")
@Data
public class LoggerProperties {

    private Class<?> databaseDriver = org.h2.Driver.class;

    private String jdbcUrl = "jdbc:h2:" + Paths.get(FileUtils.getMainClasspath(), "qqrobot", "log");
}

File: qqrobot-spring-boot-starter/src/main/java/de/honoka/qqrobot/starter/config/LoggerProperties.kt
package de.honoka.qqrobot.starter.config

import de.honoka.sdk.util.file.FileUtils
import org.h2.Driver
import org.springframework.boot.context.properties.ConfigurationProperties
import kotlin.io.path.Path

@ConfigurationProperties("honoka.qqrobot.logger")
class LoggerProperties(
    
    var databaseDriver: Class<*> = Driver::class.java,
    
    var jdbcUrl: String = "jdbc:h2:${Path(FileUtils.getMainClasspath(), "qqrobot", "log")}"
)
